packages/aws-cdk-lib/aws-ecs/lib/base/service-managed-volume.ts

```diff
@@ -0,0 +1,310 @@
+      sourceVolume: this.name,
+      ...mountPoint,
+    });
+  }
+
+  private validateVolumeConfiguration(volumeConfig?: ServiceManagedEBSVolumeConfiguration) {
+    if (!volumeConfig) return;
+
+    const { volumeType = ec2.EbsDeviceVolumeType.GP2, iops, sizeInGiB, throughput, snapShotId } = volumeConfig;
+
+    // Validate if both sizeInGiB and snapShotId are not specified.
+    if (sizeInGiB === undefined && snapShotId === undefined) {
+      throw new Error('sizeInGiB or snapShotId must be specified');
+    }
+
+    if (snapShotId && !Token.isUnresolved(snapShotId) && !/^snap-[0-9a-fA-F]+$/.test(snapShotId)) {
+      throw new Error('`snapshotId` does match expected pattern. Expected `snap-<hexadecmial value>` (ex: `snap-05abe246af`) or Token');
+    }
+
+    // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-service-servicemanagedebsvolumeconfiguration.html#cfn-ecs-service-servicemanagedebsvolumeconfiguration-sizeingib
+    const sizeInGiBRanges = {
+      [ec2.EbsDeviceVolumeType.GP2]: { minSize: 1, maxSize: 16384 },
+      [ec2.EbsDeviceVolumeType.GP3]: { minSize: 1, maxSize: 16384 },
+      [ec2.EbsDeviceVolumeType.IO1]: { minSize: 4, maxSize: 16384 },
+      [ec2.EbsDeviceVolumeType.IO2]: { minSize: 4, maxSize: 16384 },
+      [ec2.EbsDeviceVolumeType.SC1]: { minSize: 125, maxSize: 16384 },
+      [ec2.EbsDeviceVolumeType.ST1]: { minSize: 125, maxSize: 16384 },
+      [ec2.EbsDeviceVolumeType.STANDARD]: { minSize: 1, maxSize: 1024 },
+    };
+
+    // Validate volume sizeInGiB ranges.
+    if (sizeInGiB !== undefined) {
+      const { minSize, maxSize } = sizeInGiBRanges[volumeType];
+      if (sizeInGiB < minSize || sizeInGiB > maxSize) {
+        throw new Error(`'${volumeType}' volumes must have a size between ${minSize} and ${maxSize} GiB, got ${sizeInGiB} GiB`);
+      }
+    }
+
+    // Validate throughput.
+    // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-service-servicemanagedebsvolumeconfiguration.html#cfn-ecs-service-servicemanagedebsvolumeconfiguration-throughput
+    if (throughput !== undefined) {
+      if (volumeType !== ec2.EbsDeviceVolumeType.GP3) {
+        throw new Error(`'throughput' can only be configured with gp3 volume type, got ${volumeType}`);
+      } else if (throughput > 1000) {
+        throw new Error(`'throughput' must be less than or equal to 1000 MiB/s, got ${throughput} MiB/s`);
+      }
+    }
+
+    // Check if IOPS is not supported for the volume type.
+    // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-service-servicemanagedebsvolumeconfiguration.html#cfn-ecs-service-servicemanagedebsvolumeconfiguration-iops
+    if ([ec2.EbsDeviceVolumeType.SC1, ec2.EbsDeviceVolumeType.ST1, ec2.EbsDeviceVolumeType.STANDARD,
+      ec2.EbsDeviceVolumeType.GP2].includes(volumeType) && iops !== undefined) {
+      throw new Error(`'iops' cannot be specified with '${volumeType}' volume type`);
```

Comment:
With this message, the user would first specify STANDARD and encounter an error, then maybe GP2 with iops specified. In that case, they would encounter the same error again.
Therefore, I think it would be more helpful to include all four unsupported volume types in the message. (Fortunately, it's not that many words.)